Sources/Transport/TCP/TCPTransport.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  TCPTransport.swift
+//  StatsdClient
+//
+//  Created by Khoi Lai on 10/11/17.
+//  Copyright © 2017 StatsdClient. All rights reserved.
+//
+
+import Foundation
+import CocoaAsyncSocket
+
+public class TCPTransport: NSObject, Transport {
+    let host: String
+    let port: UInt16
+
+    private var completionBlocks = [Int: TransportCompletionCallback]()
+    private var tag: Int = 0
+
+    private lazy var socket: GCDAsyncSocket = {
+        let socket = GCDAsyncSocket(delegate: self,
```

Comment:
I suggest returning GCDAsyncSocket's instance intermediately. Don't need to `let socket = `

Reply:
ah true,